File: app/src/main/java/com/camp/campingtripplanner/SelectTripViewModel.java
package com.camp.campingtripplanner;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

public class SelectTripViewModel extends ViewModel {
    private LiveData<List<Trip>> trips;

    public SelectTripViewModel(AppDatabase appDatabase){
        trips = appDatabase.tripDao().getAll();
    }

    public LiveData<List<Trip>> getTrips(){
        return trips;
    }
}

File: app/src/main/java/com/camp/campingtripplanner/ui/selecttrip/SelectTripViewModel.kt
package com.camp.campingtripplanner.ui.selecttrip

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.camp.campingtripplanner.CampingTrip
import com.camp.campingtripplanner.CampingTripRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOn
import kotlinx.coroutines.launch
import javax.inject.Inject
import kotlin.coroutines.CoroutineContext

@HiltViewModel
class SelectTripViewModel @Inject constructor(
    private val tripRepository: CampingTripRepository
) : ViewModel() {

    var campingTrips = listOf<CampingTrip>()

    init {
        getTrips()
    }

    fun getTrips() {
//        return tripRepository.getAllCampingTrips()
//            .flowOn(Dispatchers.IO)
//            .collect {
//                campingTrips = it
//            }
        viewModelScope.launch {
            tripRepository.getAllCampingTrips().collect {
                campingTrips = it
            }
        }
    }
}
